Firestore/core/test/firebase/firestore/model/transform_operations_test.cc

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/model/transform_operations.h"
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+class DummyOperation : public TransformOperation {
+ public:
+  DummyOperation() {
+  }
+
+  Type type() const override {
+    return static_cast<Type>(-1);
```

Comment:
Converting an `int` to an enum when the integer has a value which is not in the enum results in [undefined behavior](https://stackoverflow.com/questions/33607809/can-an-out-of-range-enum-conversion-produce-a-value-outside-the-underlying-type).

Reply:
fixed

Reply:
Sorry, I misread the link I posted. When it speaks about "the range of the enumeration values", it means the range of values that the underlying type of the enum can hold, not the values of enum members. In other words:
```cpp
enum class Foo : unsigned int { SomeValue = 1};
Foo x = static_cast<Foo>(42); // Well-defined, because a small positive value is within the range of unsigned int
Foo y = static_cast<Foo>(-42); // Undefined behavior, because a negative value is outside the range of unsigned int
```

Sorry about the trouble, feel free to revert back.